File: EquipmentLoan.Domain/Entities/Loan.cs
using EquipmentLoan.Domain.Core;
using System.Text.Json.Serialization;

namespace EquipmentLoan.Domain.Entities;

public class Loan : BaseEntity
{
    public int EquipmentId { get; set; }
    [JsonIgnore]
    public Equipment Equipment { get; set; } = null!;

    public int EmployeeId { get; set; }
    [JsonIgnore]
    public Employee Employee { get; set; } = null!;

    public DateTime LoanDate { get; set; } = DateTime.UtcNow;
    public DateTime ExpectedReturnDate { get; set; }
    public DateTime? ActualReturnDate { get; set; }
    public string Status { get; set; } = "Active";
    public string? Notes { get; set; }

    public int? ApprovedByUserId { get; set; }  //la logica es que tome el PK asignada segun el usuario 
                                              // haciendo uso de su valor numerico en la DB por medio [int?
    [JsonIgnore]
    public User? ApprovedByUser { get; set; } //la logica es que tome el PK asignada segun el usuario 
                                              // haciendo uso de su valor numerico en la DB por medio [int?
}

File: frontend/src/types/loan.ts
export interface LoanDto {
  id: number;
  equipmentId: number;
  equipmentName: string;
  employeeId: number;
  employeeName: string;
  loanDate: string;
  expectedReturnDate: string;
  actualReturnDate?: string | null;
  status: string;
  notes?: string | null;
}

export interface CreateLoanDto {
  equipmentId: number;
  employeeId: number;
  expectedReturnDate: string;
  notes?: string | null;
  approvedByUserId?: number | null;
}

export interface ReturnLoanDto {
  actualReturnDate?: string | null;
  notes?: string | null;
}
